src/components/Homepage.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+
+import { login } from '../actions';
+
+import LoginForm from './LoginForm';
+
+class Homepage extends React.Component {
+
+  handleSubmit = (formData) => {
+    const {username, password} = formData;
+    this.props.login(username, password);
+  }
+
+  render() {
+    if (this.props.loggedIn) {
+      return <Redirect to="/preferences" />
```

Comment:
Keep in mind that this only occurs the first time. I would redirect to `/` and then there (`Authenticated` component), if the user miss some config, you always kick it back to `/preferences`